File: src/App/Platforms/Android/Accessibility/Browser.cs
using System;

namespace Bit.Droid.Accessibility
{
    public class Browser
    {
        public Browser(string packageName, string uriViewId)
        {
            PackageName = packageName;
            UriViewId = uriViewId;
        }

        public Browser(string packageName, string uriViewId, Func<string, string> getUriFunction)
            : this(packageName, uriViewId)
        {
            GetUriFunction = getUriFunction;
        }

        public string PackageName { get; set; }
        public string UriViewId { get; set; }
        public Func<string, string> GetUriFunction { get; set; } = (s) => s;
    }
}

File: app/src/main/kotlin/com/x8bit/bitwarden/data/autofill/accessibility/model/Browser.kt
package com.x8bit.bitwarden.data.autofill.accessibility.model

/**
 * A model representing a supported browser.
 */
data class Browser(
    val packageName: String,
    val possibleUrlFieldIds: List<String>,
    val urlExtractor: (String) -> String? = { it },
) {
    constructor(
        packageName: String,
        urlFieldId: String,
        urlExtractor: (String) -> String? = { it },
    ) : this(
        packageName = packageName,
        possibleUrlFieldIds = listOf(urlFieldId),
        urlExtractor = urlExtractor,
    )
}
